extras/mmu_machine.py

```diff
@@ -552,8 +587,10 @@ def sync(self, new_sync_mode):
             s_kinematics = ffi_main.gc(s_alloc, ffi_lib.free)
             self._prev_sk.append(s.set_stepper_kinematics(s_kinematics))
             self._prev_rd.append(s.get_rotation_distance()[0])
-            following_toolhead.step_generators.remove(s.generate_steps)
-            driving_toolhead.register_step_generator(s.generate_steps)
+            if hasattr(following_toolhead, 'step_generators'):
+                following_toolhead.step_generators.remove(s.generate_steps)
+            if hasattr(driving_toolhead, 'register_step_generator'):
+                driving_toolhead.register_step_generator(s.generate_steps)
```

Comment:
I suspect we need to do more than this...effectively this is just removing these lines of code for newer klipper.